operator_ui/src/components/Button.tsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react'
+import {
+  createStyles,
+  Theme,
+  withStyles,
+  WithStyles
+} from '@material-ui/core/styles'
+import { PropTypes as MuiPropTypes } from '@material-ui/core'
+import MuiButton, {
+  ButtonProps as MuiButtonProps
+} from '@material-ui/core/Button'
+import { TouchRippleProps } from '@material-ui/core/ButtonBase/TouchRipple'
+import classNames from 'classnames'
+
+const styles = (theme: Theme) =>
+  createStyles({
+    default: {
+      borderColor: '#BECAD6',
+      '&:hover': {
+        backgroundColor: theme.palette.common.white,
+        borderColor: '#BECAD6',
+        boxShadow:
+          '0 2px 4px 0 rgba(0,123,255,0.06), 0 2px 2px 0 rgba(0,0,0,0.06)'
+      }
+    },
+    primary: {
+      boxShadow: '0 0',
+      backgroundColor: theme.palette.primary.main,
+      color: theme.palette.common.white,
+      '&:hover': {
+        backgroundColor: theme.palette.primary.main,
+        boxShadow:
+          '0 2px 4px 0 rgba(0,123,255,0.19), 0 2px 2px 0 rgba(0,0,0,0.15)'
+      }
+    },
+    secondary: {
+      '&:hover': {
+        backgroundColor: theme.palette.common.white
+      }
+    },
+    danger: {
+      borderColor: theme.palette.error.main,
+      color: theme.palette.error.main,
+      '&:hover': {
+        backgroundColor: theme.palette.common.white,
+        borderColor: theme.palette.error.main,
+        boxShadow:
+          '0 2px 4px 0 rgba(0,123,255,0.06), 0 2px 2px 0 rgba(0,0,0,0.06)'
+      }
+    },
+    defaultRipple: {
+      color: '#818EA3'
+    }
+  })
+
+// Unfortunately @material-ui/core does not export the type so we have to redefine it.
+type MuiButtonVariant =
+  | 'text'
+  | 'flat'
+  | 'outlined'
+  | 'contained'
+  | 'raised'
+  | 'fab'
+  | 'extendedFab'
+  | undefined
+
+export type ButtonVariant =
+  | MuiButtonVariant
+  | 'primary'
+  | 'secondary'
+  | 'danger'
+  | 'default'
+
+const muiProps = (variant: ButtonVariant, classes: any): MuiButtonProps => {
```

Comment:
@smartcontractkit/core-devs any idea how to extract this type annotation?

https://github.com/mui-org/material-ui/blob/master/packages/material-ui/src/styles/withStyles.d.ts#L40

Reply:
@rupurt hmm, are you asking how ClassNameMap's typed? https://github.com/mui-org/material-ui/blob/master/packages/material-ui-styles/src/withStyles/withStyles.d.ts#L68 
If MUI doesn't export the needed annotations, we can duplicate definitions from their codebase

Reply:
@NavyAdmiral I'm wondering if there's any way around duplicating it. Like is there a special escape hatch. We're on `v3.9.2` which is different to the latest `v4.2.1` and I'd prefer not to try and keep them in sync. 

What's your thoughts on just leaving it as `any` with a comment if there is no way to pull it from the library?

Reply:
Historically, if these types didn't change much across MUI versions, then we could use `patch-package` and export the needed type manually. Personally, I'd be good with just leaving it as `any`

Reply:
👍 `patch-package` seems pretty cool. Normally I'd just maintain a fork until it's been merged upstream but this is another good option.

Reply:
@rupurt did you make up your mind about this? I'm a passionate user of `patch-package`, but this usecase might be a bit contrived.

Reply:
@NavyAdmiral going with `any`